### ISSUE TITLE
Editing the user profile: autocomplete freetext limitation

### ISSUE THREAD

**Issue by ekhattar** [gabriela99/capstone-code#4]
## Description

When a user is editing their profile, the fields "role" and "skills" are currently free-text with an autocomplete function.

This means when a user starts typing, they see **predefined** options from the role and skills lists (`roleOptions` and `skillOptions`).

https://github.com/gabriela99/capstone-code/blob/b33ae0e043e07da79513359108b3aa90835e7a7e/app/src/main/java/com/example/capstone_code/SettingsActivity.java#L47-L48

Right now, when the user clicks save, the information is saved by calling `saveUserInformation()`
https://github.com/gabriela99/capstone-code/blob/b33ae0e043e07da79513359108b3aa90835e7a7e/app/src/main/java/com/example/capstone_code/SettingsActivity.java#L83-L88

And in `saveUserInformation()`, the field values are retrieved before being stored:
https://github.com/gabriela99/capstone-code/blob/b33ae0e043e07da79513359108b3aa90835e7a7e/app/src/main/java/com/example/capstone_code/SettingsActivity.java#L156-L159

Because it is a **free-text field** and the autocomplete shows suggestions, **it is necessary to limit what a user can choose to the predefined options**

This can be done by validating the user input before saving in the `setOnClickListener`, e.g.,

```java
        confirmSettings.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // Check if the inputs are valid before saving
                boolean isInputValid = validateInput();
                if (isInputValid) {
                    saveUserInformation();
                } else {
                     // Show an error message if the inputs are not valid
                    Toast.makeText(SettingsActivity.this,"Invalid Input! Please check that selected role and skills match the available options.",Toast.LENGTH_SHORT).show();
                }

            }
        });
```

Where `validateInput()` would:
1. retrieve the input values the same way it is done now in `saveUserInformation()` https://github.com/gabriela99/capstone-code/blob/b33ae0e043e07da79513359108b3aa90835e7a7e/app/src/main/java/com/example/capstone_code/SettingsActivity.java#L158-L159
2. check if the input values are in the predefined lists  https://github.com/gabriela99/capstone-code/blob/b33ae0e043e07da79513359108b3aa90835e7a7e/app/src/main/java/com/example/capstone_code/SettingsActivity.java#L47-L48

For step 2, this means for any given string `Abc` to check if it is in the string array `Abc, Def, Xyz`.

The easiest way to do this is to iterate through the string array `Abc, Def, Xyz` with a `for` loop to check if the user input value equals any of the array items. Since we have two fields we want to do this check for, this logic should be **re-usable** (its own function):

```java
    private boolean isStringInList(String[] listOptions, String userInput) {
        boolean elementFound = false;
        // Check an item is in an array https://www.geeksforgeeks.org/check-if-a-value-is-present-in-an-array-in-java/
        for (String validItem : listOptions) {
            if (userInput.equalsIgnoreCase(validItem)) {
                elementFound = true;
            }
        }
        return elementFound;
    }
```

⚠️  Note that it is **important** to use `String. equalsIgnoreCase()` because this will check if a user inputs `agile` is equal to the predefined item `Agile` (which should be true).

Then the validator `validateInput()` could call `isStringInList()` for both inputs separately, and return `true` (settings are valid) or `false` (settings are invalid).

```java
    // Checks that user input conforms to the list of options: true if valid input, false if invalid
    private boolean validateInput() {
        // Retrieve what the user inputted into the fields
        role = mRole.getText().toString();
        skills = mSkills.getText().toString();
        // Do the check for both fields
        boolean isRoleValid = isStringInList(roleOptions, role);
        boolean isSkillsValid = isStringInList(skillOptions, skills);
        // If both role and skills are from the predefined list, return true for valid settings
        if (isRoleValid && isSkillsValid) {
            return true;
        } else {
            return false;
        }
    }
```

To summarize:

1. Before calling `saveUserInformation()`, the user inputs for `role` and `skills` should be validated
2. `validateInput()` gets the values of both fields, then checks for each value if they are in the predefined lists using `isStringInList()`
3. If both the user inputs are valid (so are from the lists), then the validator returns `true` and the user information should be saved
4. If either one or both are invalid, then a toast is shown

Note: to show the toast message, make sure to import it `import android.widget.Toast;` at the top of the `SettingsActivity.java` file

**Comment by gabriela99** [comment]
Implemented all of this, done.
